test/library/packages/LinearAlgebra/performance/me/me-perf.chpl

```diff
@@ -0,0 +1,295 @@
+    writeln("");
+
+    t.clear();
+  }
+
+
+  {
+    // Input Matrix of Format
+    // [-2.0  1.0  0.0]
+    // [-1.0 -2.0  1.0]
+    // [ 0.0 -1.0 -2.0]
+
+    var X: [Space] eltType;
+    setDiag(X, 0, -2.0);
+    setDiag(X, 1, 1.0);
+    setDiag(X, -1, -1.0);
+
+    var M: [X.domain] eltType;
+
+    for 1..iters {
+      t.start();
+      M = expm(X);
+      t.stop();
+    }
+
+    if !correctness then
+      writeln('LinearAlgebra.expm_edge: ' + ((t.elapsed()/iters): string));
+    else
+      writeln(M);
+    writeln("");
+
+    t.clear();
+  }
+
+  {
+    // Testing eltType=Complex: Input Matrix of Format
+    // [ 1.0 + 1i 1.0       0.0     ]
+    // [-1.0      1.0 + 1i  1.0     ]
+    // [ 0.0     -1.0       1.0 + 1i]
+
+    var X: [Space] complex;
+    setDiag(X, 0, 1.0 + 1i);
+    setDiag(X, 1, 1.0);
+    setDiag(X, -1, -1.0);
+
+    var M: [X.domain] complex;
+
+    for 1..iters {
+      t.start();
+      M = expm(X);
+      t.stop();
+    }
+
+    if !correctness then
+      writeln('LinearAlgebra.expm_complex: ' + ((t.elapsed()/iters): string));
+    else
+      writeln(M);
+    writeln("");
+
+    t.clear();
+  }
+}
+
+proc test_sinm() {
+  {
+    // Testing eltType=Complex: Input Matrix of Format
+    // [ 1.0 + 1i 1.0       0.0     ]
+    // [-1.0      1.0 + 1i  1.0     ]
+    // [ 0.0     -1.0       1.0 + 1i]
+
+    const Space = {0..<m, 0..<m};
+    var t: Timer;
+
+    var X: [Space] complex;
+    setDiag(X, 0, 1.0 + 1i);
+    setDiag(X, 1, 1.0);
+    setDiag(X, -1, -1.0);
+
+    var M: [X.domain] complex;
+
+    for 1..iters {
+      t.start();
+      M = sinm(X);
+      t.stop();
+    }
+
+    if !correctness then
+      writeln('LinearAlgebra.sinm: ' + ((t.elapsed()/iters): string));
+    else
+      writeln(M);
+    writeln("");
+
+    t.clear();
+  }
+}
+
+proc test_cosm() {
+  {
+    // Testing eltType=Complex: Input Matrix of Format
+    // [ 1.0 + 1i 1.0       0.0     ]
+    // [-1.0      1.0 + 1i  1.0     ]
+    // [ 0.0     -1.0       1.0 + 1i]
+
+    const Space = {0..<m, 0..<m};
+    var t: Timer;
+
+    var X: [Space] complex;
+    setDiag(X, 0, 1.0 + 1i);
+    setDiag(X, 1, 1.0);
+    setDiag(X, -1, -1.0);
+
+    var M: [X.domain] complex;
+
+    for 1..iters {
+      t.start();
+      M = cosm(X);
+      t.stop();
+    }
+
+    if !correctness then
+      writeln('LinearAlgebra.cosm: ' + ((t.elapsed()/iters): string));
+    else
+      writeln(M);
+    writeln("");
+
+    t.clear();
+  }
+}
+
+proc test_sincos() {
+  {
+    // Testing eltType=Complex: Input Matrix of Format
+    // [ 1.0 + 1i 1.0       0.0     ]
+    // [-1.0      1.0 + 1i  1.0     ]
+    // [ 0.0     -1.0       1.0 + 1i]
+
+    const Space = {0..<m, 0..<m};
+    var t: Timer;
+
+    var X: [Space] complex;
+    setDiag(X, 0, 1.0 + 1i);
+    setDiag(X, 1, 1.0);
+    setDiag(X, -1, -1.0);
+
+    var Z_sin: [X.domain] complex;
+    var Z_cos: [X.domain] complex;
+
+    for 1..iters {
+      t.start();
+      (Z_sin, Z_cos) = sincos(X);
```

Comment:
Probably not a serious issue, but there maybe some costs associated with unpacking the return tuple here. 

This might be a serious issue especially with very small matrices.